packages/playwright-test/src/reporters/azure.ts

```diff
@@ -0,0 +1,475 @@
+      disabled: colors.gray(`Test ${test.title} - ${testResult.status}`),
+      timedOut: colors.gray(`Test ${test.title} - ${testResult.status}`)
+    };
+    if (testResult.status)
+      this.log(map[testResult.status]);
+
+  }
+
+  private async createRun(runName: string): Promise<TestInterfaces.TestRun | void> {
+    try {
+      await this.checkProject(this.projectName);
+      const runModel: TestInterfaces.RunCreateModel = {
+        name: runName,
+        automated: true,
+        configurationIds: [1],
+        plan: { id: String(this.planId) }
+      };
+      if (!this.testApi)
+        this.testApi = await this.connection.getTestApi();
+      const adTestRun = await this.testApi.createTestRun(runModel, this.projectName);
+      if (adTestRun)
+        return adTestRun;
+      else
+        throw new Error('Failed to create test run');
+
+    } catch (e) {
+      this.warning(colors.red(e.message));
+      this.isDisabled = true;
+    }
+  }
+
+  private removePublished(testAlias: string): void {
+    const resultIndex = this.resultsToBePublished.indexOf(testAlias);
+    if (resultIndex !== -1)
+      this.resultsToBePublished.splice(resultIndex, 1);
+  }
+
+  private async checkProject(projectName: string): Promise<TeamProject | void> {
+    try {
+      if (!this.coreApi)
+        this.coreApi = await this.connection.getCoreApi();
+      const project = await this.coreApi.getProject(projectName);
+      if (project)
+        return project;
+      else
+        throw new Error(`Project ${projectName} does not exist. Reporting is disabled.`);
+
+    } catch (e) {
+      this.warning(colors.red(e.message));
+      this.isDisabled = true;
+    }
+  }
+
+  private async getTestPointIdsByTCIds(planId: number, testcaseIds: number[]): Promise<number[]> {
+    const pointsIds: number[] = [];
+    try {
+      const pointsQuery: TestInterfaces.TestPointsQuery = {
+        pointsFilter: { testcaseIds }
+      };
+      if (!this.testApi)
+        this.testApi = await this.connection.getTestApi();
+      const pointsQueryResult: TestInterfaces.TestPointsQuery = await this.testApi.getPointsByQuery(
+          pointsQuery,
+          this.projectName
+      );
+      if (pointsQueryResult.points) {
+        pointsQueryResult.points.forEach((point: TestPoint) => {
+          if (point.testPlan && point.testPlan.id && parseInt(point.testPlan.id, 10) === planId)
+            pointsIds.push(point.id);
+          else
+            throw new Error(`Could not find test point for test case [${point.testCase.id}] associated with test plan ${this.planId}. Check, maybe testPlanId, what you specifiyed, is incorrect.`);
+        });
+      }
+      return pointsIds;
+    } catch (e) {
+      this.log(colors.red(e.message));
+    }
+    return pointsIds;
+  }
+
+  private addReportingOverride = (api: Test.ITestApi): Test.ITestApi => {
+    // https://github.com/microsoft/azure-devops-node-api/issues/318#issuecomment-498802402
```

Comment:
having a link is a good start, but lets also add a few words why we need it exactly before the link.